packages/pg/lib/crypto/crypto-shim.js

```diff
@@ -0,0 +1,101 @@
+// MIT License
+// Copyright (c) 2022 - 2023 Neon Inc.
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+const sha256Sync = require('./sha256')
+
+/**
+ * Patch the Node.js crypto APIs to work outside Node.js
+ */
+module.exports = function patchCrypto(crypto) {
+  crypto.randomBytes = function (length) {
+    return globalThis.crypto.getRandomValues(Buffer.alloc(length))
+  }
+
+  crypto.createHash = function (type) {
+    if (type !== 'sha256') throw new Error('Only sha256 is supported')
+    return {
+      update: function (data) {
+        return {
+          digest: function () {
+            return Buffer.from(sha256Sync(data))
+          },
+        }
+      },
+    }
+  }
+
+  crypto.createHmac = function createHmac(type, key) {
+    if (type !== 'sha256') throw new Error('Only sha256 is supported')
+    return {
+      update: function (data) {
+        return {
+          digest: function () {
+            if (typeof key === 'string') key = new TextEncoder().encode(key)
+            if (typeof data === 'string') data = new TextEncoder().encode(data)
```

Comment:
Nit: there's no reason to create two separate `TextEncoder` instances here. Just create one at top level and reuse it.

Reply:
Yep!